src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -636,19 +623,22 @@ private static int GetPreambleLength(byte[] data, int dataLength, Encoding encod
             }
```

Comment:
The `default` case here should take the ArraySegment, since Offset is lost on this call.